browse/static/js/catalyzex.js

```diff
@@ -0,0 +1,60 @@
+(async () => {
```

Comment:
@riltonfranzonee what's the reasoning behind this being async?

Reply:
It's because we are using promises to perform asynchronous tasks (i.e. retrieving data from CatalyzeX). In line 35 you can see where exactly we are calling the promise:

![Screen Shot 2023-01-03 at 16 53 42](https://user-images.githubusercontent.com/58868651/210431256-6c415d86-191e-459a-bd96-69160eb7371e.png)

Reply:
Note: From seeing some of the other JS files in `static/js/`, there's an approach to just having it be a function() while using `await` and using $.get (which uses no timeout) rather than `$.ajax` with the 2s timeout that we're using. So there's a way to switch to the other style if that's a preferred convention, but it will not have a timeout included.

Reply:
Gotcha. Recently [Dagshub](https://dagshub.com/) integrated their system with Arxiv and they also used `await` (and even `fetch` for making the API calls), so perhaps this wouldn't be an issue for us?

```
	(async () => {
		let response = await fetch(dagshubRepositoryFromPaperApi);
		if (!response.ok) {
		  console.error(`Unable to fetch data from ${dagshubRepositoryFromPaperApi}`)
		  render({"DagsHub is not available": "an error occurred while fetching"});
		  return;
		}
		const dagshub_data = await response.json();
		render(dagshub_data);
	  })()
```

The great thing about the `async-await` syntax is that the code looks cleaner and requires fewer callbacks :)

Reply:
yes, a couple others are using async and awaits too :) although I believe Himanshu's referring to the first line of each of these JS files, which start with `(function()` rather than `(async () => {`

Reply:
Ah, I see. Indeed they all seem to start with `(function() {`. 

In practical terms, there's no difference, besides the syntax being more abbreviate and allowing us to more easily handle asynchronous calls when starting with `async () =>`. 

If you see the other implementations, for instance, they either tend to have more callback functions or they end up having to create other [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) functions inside the main IIFE just to have the async feature, which makes the code a little bit harder to read.

But it's up to you, guys – if you prefer the other syntax or feel like that would be better to get it reviewed more quickly, I can modify this. 😃

Reply:
Got it, sounds good :) Let's keep it as is unless they want us to follow that pattern.